src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs

```diff
@@ -68,32 +68,52 @@ public static IResourceBuilder<PostgresDatabaseResource> AddDatabase(this IResou
     {
-        if (builder.ApplicationBuilder.Resources.OfType<PgAdminContainerResource>().Any())
+        if (builder.ApplicationBuilder.Resources.OfType<PgAdminContainerResource>().SingleOrDefault() is { } existingPgAdminResource)
         {
+            var builderForExistingResource = builder.ApplicationBuilder.CreateResourceBuilder(existingPgAdminResource);
+            configureContainer?.Invoke(builderForExistingResource);
             return builder;
         }
+        else
+        {
+            builder.ApplicationBuilder.Services.TryAddLifecycleHook<PgAdminConfigWriterHook>();
 
-        builder.ApplicationBuilder.Services.TryAddLifecycleHook<PgAdminConfigWriterHook>();
+            containerName ??= $"{builder.Resource.Name}-pgadmin";
 
-        containerName ??= $"{builder.Resource.Name}-pgadmin";
+            var pgAdminContainer = new PgAdminContainerResource(containerName);
+            var pgAdminContainerBuilder = builder.ApplicationBuilder.AddResource(pgAdminContainer)
+                                                 .WithImage(PostgresContainerImageTags.PgAdminImage, PostgresContainerImageTags.PgAdminTag)
+                                                 .WithImageRegistry(PostgresContainerImageTags.PgAdminRegistry)
+                                                 .WithHttpEndpoint(targetPort: 80, name: "http")
+                                                 .WithEnvironment(SetPgAdminEnvironmentVariables)
+                                                 .WithBindMount(Path.GetTempFileName(), "/pgadmin4/servers.json")
+                                                 .ExcludeFromManifest();
 
-        var pgAdminContainer = new PgAdminContainerResource(containerName);
-        builder.ApplicationBuilder.AddResource(pgAdminContainer)
-                                  .WithImage("dpage/pgadmin4", "8.3")
-                                  .WithImageRegistry(PostgresContainerImageTags.Registry)
-                                  .WithHttpEndpoint(targetPort: 80, port: hostPort, name: containerName)
-                                  .WithEnvironment(SetPgAdminEnvironmentVariables)
-                                  .WithBindMount(Path.GetTempFileName(), "/pgadmin4/servers.json")
-                                  .ExcludeFromManifest();
+            configureContainer?.Invoke(pgAdminContainerBuilder);
+        }
 
         return builder;
     }
 
+    /// <summary>
+    /// Configures the host port that the PGAdmin resource is exposed on instead of using randomly assigned port.
+    /// </summary>
+    /// <param name="builder">The resource builder for PGAdmin.</param>
+    /// <param name="port">The port to bind on the host. If null is used random port will be assigned.</param>
+    /// <returns>The resource builder for PGAdmin.</returns>
+    public static IResourceBuilder<PgAdminContainerResource> UseHostPort(this IResourceBuilder<PgAdminContainerResource> builder, int? port)
```

Comment:
This is new right? Are we sure `Use` is the right prefix here? Seems odd that this method is only available for this resource type.

Reply:
It's what we do with the other azure resources that have emulators:

https://github.com/dotnet/aspire/blob/06d8886d8acf0e0cb4090f15c799545a5514af9c/src/Aspire.Hosting.Azure.Storage/AzureStorageExtensions.cs#L149